Add Chip_seq_genomic_ranges_tutorial

vmuckerson opened the pull request:
Added a tutorial for aligning ensembl db annotations and chip-seq data from ENCODE. Also includes bio frame capabilities and genome features filtering example.

ivirshup commented:
I've pushed a commit that should make the docs render

vmuckerson commented:
@ivirshup This should now be ready for review/merging!

ivirshup commented on docs/how-to/ChIP_seq_genomic_ranges_tutorial.ipynb:
```diff
@@ -0,0 +1,2733 @@
+{
```
<p>It would be nice to link to the docs for read_table here, like:</p><p> </p>

<pre class="ql-syntax" spellcheck="false">{func}<code>~bioframe.read_table</code>
</pre><p>You'll also need to add this line to the <code>intersphinx_mapping</code> in <code>docs/conf.py</code> :</p>

<pre class="ql-syntax" spellcheck="false">    "bioframe": ("https://bioframe.readthedocs.io/en/stable/", None),  
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scverse/genomic-features/pull/65/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

ivirshup replied:
<pre class="ql-syntax" spellcheck="false">Line #2.    ensdb.list_columns()  # you can custom select the columns output if exon, protein, or sequence information that is not included by default
</pre><p>I can't seem to paste an image here, but this comment is difficult to read when rendered (see <a href="https://genomic-features--65.org.readthedocs.build/en/65/how-to/ChIP_seq_genomic_ranges_tutorial.html#download-ensembl-annotations-using-genomic-features" target="_blank">here</a>). Can it be broken up or written in markdown text?</p><p> </p><p>I think the output of this line should also probably be show if this line is included.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scverse/genomic-features/pull/65/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

ivirshup replied:
<pre class="ql-syntax" spellcheck="false">Line #6.        # cols=["gene_id", "gene_name","gene_biotype", "gene_seq_start",
</pre><p>Maybe can remove these commented out lines?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scverse/genomic-features/pull/65/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

ivirshup replied:
<p>Would be nice to reference bioframe docs on <code>assembly_info</code> here </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scverse/genomic-features/pull/65/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

lauradmartens replied:
<p>very minor, but I think it should be homo sapiens</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scverse/genomic-features/pull/65/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

lauradmartens replied:
<p>maybe have the column names formatted like <code>chrom</code>, <code>start</code>,<code>end</code>. Also very minor but I think it's ascribed instead of asscribed      </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scverse/genomic-features/pull/65/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

lauradmartens replied:
<p>I guess it should be something like: "we need to merge the two dataframes" or "the two dataframes need to be merged"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scverse/genomic-features/pull/65/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

lauradmartens replied:
<p>the description does not really fit the next step I think? Here you are doing <code>overlap</code> not <code>closest</code>   <span class="ql-cursor">﻿</span></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scverse/genomic-features/pull/65/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

ivirshup commented on docs/how-to/ChIP_seq_genomic_ranges_tutorial.ipynb:
```diff
@@ -0,0 +1,2733 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #1.    genes["seq_name"] = genes["seq_name"].map(hg38.alias_dict)
</pre><p>With <code>pd.Series.map</code> any value in the series that isn't in the dict will be mapped to a null value.</p><p> </p><p>Should this be:</p><p> </p>

<pre class="ql-syntax" spellcheck="false">genes["seq_name"] = genes["seq_name"].replace(hg38.alias_dict)
</pre><p>instead? Or should non-standard contigs even be filtered out?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scverse/genomic-features/pull/65/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

lauradmartens replied:
<p>They would probably anyways be filtered out in the next step, if they do not overlap?</p>

vmuckerson replied:
<p>The unmapped chromosome calls are given NA values that are then filtered using the dropna() function.</p>

ivirshup commented on docs/how-to/ChIP_seq_genomic_ranges_tutorial.ipynb:
```diff
@@ -0,0 +1,2733 @@
+{
```
<p>How difficult would it be to plot a gene and associated peak? Like with <code>bioframe.plot_intervals</code> or <code>gos</code> ?</p><p> </p><p>Not a requirement, but I think could be nice! </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scverse/genomic-features/pull/65/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

vmuckerson replied:
I'm planning on doing this in the single-cell 10X Visium Multiome (atac-seq was mentioned in one of the issues I think so wanted to tackle that specifically) tutorial I'm working on now! Should hopefully have this for you by Monday, if all things go to plan

vmuckerson replied:
<p>Monday may have been ambitious but I'm working on this*</p>

ivirshup reviewed (commented):
Thanks for putting this together!

I've commented with a few points, mainly on some style/ technical parts.

@emdann or @lauradmartens, would you mind giving this a once over?

lauradmartens commented:
<p>Thanks a lot for the great contribution! I left a few comments!</p>

vmuckerson commented:
Thank you both for reviewing @ivirshup @lauradmartens ! I hope I addressed all of your comments, please let me know if there's anything else that should be changed or if anything remains unclear.

ivirshup reviewed (approved):
LGTM
